src/extensions/default/AutoUpdate/StateHandler.js

```diff
@@ -0,0 +1,205 @@
+
+        _file.exists(function (err, exists) {
+            if (exists) {
+                result.resolve();
+            } else {
+                result.reject();
+            }
+        });
+
+        return result.promise();
+    };
+
+    /**
+     * Parses the JSON file, and maintains a state for the parsed data
+     * @returns {$.Deferred} - a jquery deferred promise,
+     *                       that is resolved with a parsing success or failure
+     */
+    StateHandler.prototype.parse = function () {
+        var result = $.Deferred(),
+            _file = FileSystem.getFileForPath(this.path);
+        var self = this;
+
+        this.exists()
+            .done(function () {
+                FileUtils.readAsText(_file)
+                    .done(function (text) {
+                        try {
+                            if (text) {
+                                self.state = JSON.parse(text);
+                                result.resolve();
+                            } else {
+                                result.reject(FILE_READ_FAIL);
+                            }
+                        } catch (error) {
+                            result.reject(FILE_PARSE_EXCEPTION);
+                        }
+                    })
+                    .fail(function () {
+                        result.reject(FILE_NOT_READ);
+                    });
+
+            })
+            .fail(function () {
+                result.reject(FILE_NOT_FOUND);
+            });
+
+        return result.promise();
+    };
+
+    /**
+     * Sets the value of a key in a json file.
+     * @param   {string} key  - key for which the value is to be set
+     * @param   {string} value - the value to be set for the given key
+     * @returns {$.Deferred} - a jquery deferred promise, that is resolved with a write success or failure
+     */
+    StateHandler.prototype.set = function (key, value) {
+        this.state = this.state || {};
+        this.state[key] = value;
+
+        return this.write(true);
+    };
+
+    /**
+     * Gets the value for a given key, from the in-memory state maintained for a json file.
+     * @param   {string} key - key for which value is to be retrieved
+     * @returns {string} value for the given key
+     */
+    StateHandler.prototype.get = function (key) {
+        var retval = null;
+
+        if (this.state && this.state[key]) {
+            retval = this.state[key];
+        }
+
+        return retval;
+    };
+
+
+    /**
+     * Performs the write of JSON object to a file.
+     * @param   {string} filepath - path to JSON file
+     * @param   {object} json  - JSON object to write
+     * @returns {$.Deferred} - a jquery deferred promise,
+     *                       that is resolved with the write success or failure
+     */
+    function _write(filePath, json) {
+        var result = $.Deferred(),
+            _file = FileSystem.getFileForPath(filePath);
```

Comment:
check for validity of `_file`

Reply:
I think there is not harm in checking for validity of `_file` given the fact that you are going to `stringify` the JSON irrespective of whether `_file` is valid or not.

Reply:
@nethip , the requirement here is to do a force write. If the file exists, the content is overwritten. If it doesn't exist, the file is created and written. I am passing the boolean to be "true" in writeText, which ensures this.
Hence, the validity is not needed here.

Reply:
@mbhavi Let us not have discussion of having a null check or not. It is a simple `null` check. If you still feel that is not required, post merging this PR, one of us will add it.

Reply:
@nethip , addressed in 4fb6582